### PR TITLE
Suppress warnings from Qiskit Nature

### LINKED ISSUE #154: Migrate to Qiskit Nature 0.6.0
Note: We may have to wait for Quantum Serverless to accept Qiskit Nature 0.6.0 before we migrate

Migrate to Qiskit Nature 0.6.0. Resolution of this issue should result in zero deprecation warnings from Qiskit Nature during unit testing.

### PR DESCRIPTION
**Pull request by SaashaJoshi**
Resolves #154

### CODE REVIEW

File: circuit_knitting_toolbox/entanglement_forging/cholesky_decomposition.py
Comment (garrison) [inline circuit_knitting_toolbox/entanglement_forging/cholesky_decomposition.py:40]: Since it may violate a user's expectations to modify global state like this, it deserves some comments here of why it's necessary and also deserves mention in the release notes.  (At least it is a temporary thing, and will have no effect on Qiskit Nature versions 0.7 and higher.)

Comment (garrison) [reply circuit_knitting_toolbox/entanglement_forging/cholesky_decomposition.py:40]: Also, it might be better to move this one level up, to `entanglement_forging/__init__.py`, for increased visibility.

**Comment by garrison** [comment]
When I run this branch under pytest, all versions pass:

```
  py38: OK (105.50=setup[71.32]+cmd[34.19] seconds)
  py39: OK (129.52=setup[89.50]+cmd[40.02] seconds)
  py310: OK (138.50=setup[100.98]+cmd[37.52] seconds)
  py311: OK (135.71=setup[102.40]+cmd[33.30] seconds)
  congratulations :) (509.33 seconds)
```

When I run the tests again locally against current development versions, using `python tools/extremal_dependency_versions.py pin_dependencies dev --inplace` (this is destructive; it modifies `pyproject.toml`), then all tests pass again as well:

```
  py38: OK (878.60=setup[865.89]+cmd[12.71] seconds)
  py39: OK (161.98=setup[147.90]+cmd[14.08] seconds)
  py310: OK (152.29=setup[140.42]+cmd[11.88] seconds)
  py311: OK (152.60=setup[142.16]+cmd[10.44] seconds)
  congratulations :) (1345.72 seconds)
```

When I run notebooks from this branch, they fail on all versions of python:

```
  py38-notebook: FAIL code 1 (181.68=setup[73.70]+cmd[107.98] seconds)
  py39-notebook: FAIL code 1 (184.47=setup[74.37]+cmd[110.11] seconds)
  py310-notebook: FAIL code 1 (189.66=setup[75.84]+cmd[113.82] seconds)
  py311-notebook: FAIL code 1 (173.32=setup[75.39]+cmd[97.94] seconds)
  evaluation failed :( (729.22 seconds)
```

So, at the moment, it seems that the tests fail on the notebook(s) only regardless of python version.

File: circuit_knitting_toolbox/entanglement_forging/__init__.py
Comment (garrison) [inline circuit_knitting_toolbox/entanglement_forging/__init__.py:34]: These lines actually need to be moved to `circuit_knitting/forging/__init__.py` now that #244 is merged.

Comment (garrison) [reply circuit_knitting_toolbox/entanglement_forging/__init__.py:34]: Also, once we finally figure out the magic to get everything working, we should double check that this block is actually required to get tests to pass.  The [migration guide](https://qiskit.org/ecosystem/nature/migration/0.6_c_qubit_converter.html#paulisumop-vs-sparsepauliop) suggests that it is not always necessary to change this setting.  If you provide a `SparsePauliOp` to a function that returns an operator, it should result in a `SparsePauliOp` regardless of this setting, from what I understand.

Comment (SaashaJoshi) [reply circuit_knitting_toolbox/entanglement_forging/__init__.py:34]: Yes, this was in fact added after a discussion between @caleb-johnson and Max (Rossmannek). He suggested that we can suppress the warnings with these settings until opflow is dropped, which will be in the 0.8.0 release.

**Comment by garrison** [comment]
The following line in `pyproject.toml` is preventing Qiskit Nature 0.6.0 from being used in the notebook tests, but I believe that using `SparsePauliOp` (this PR) requires getting rid of that upper bound.  (At least, there is no attribute `setting.use_pauli_sum_op` in earlier versions.) That might then cause a failure related to quantum serverless in one of the notebooks, but we can evaluate what to do about that (this is all related to #108).

https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/blob/4a0cd45c5f28b196800aefd843f380ba7136494c/pyproject.toml#L84

The dependency should also be updated at the following line in order for the minimum version tests to pass:

https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/blob/4a0cd45c5f28b196800aefd843f380ba7136494c/pyproject.toml#L31

The README badge should be updated to match that version, too.

File: pyproject.toml
Comment (garrison) [inline pyproject.toml:31]: reminder to update the version badge in `README.md`, too.

**Review by garrison — DISMISSED** [review]
This looks great.  Let's wait for @caleb-johnson's approval, too.

fwiw, I tried to remove `settings.use_pauli_sum_op = False`, but tests failed, so it looks like we have to keep it.  Not a big surprise, but I wanted to double check that it is absolutely necessary.

**Comment by garrison** [comment]
Oh yes -- I forgot before approving: This certainly deserves a release note, too.

File: circuit_knitting/forging/cholesky_decomposition.py
Comment (caleb-johnson) [inline circuit_knitting/forging/cholesky_decomposition.py:22]:
```suggestion
from qiskit.quantum_info import SparsePauliOp
```

File: circuit_knitting/forging/entanglement_forging_ground_state_solver.py
Comment (caleb-johnson) [inline circuit_knitting/forging/entanglement_forging_ground_state_solver.py:36]:
```suggestion
from qiskit.quantum_info import SparsePauliOp
```

File: docs/entanglement_forging/tutorials/tutorial_2_forging_with_quantum_serverless.ipynb
Comment (caleb-johnson) [inline docs/entanglement_forging/tutorials/tutorial_2_forging_with_quantum_serverless.ipynb:14]: Nice

**Review by caleb-johnson — COMMENTED** [review]
This looks great, thanks @SaashaJoshi and @garrison !

I suggested a couple of small simplifications on the import and like Jim suggested, please add a [release note](https://docs.openstack.org/reno/latest/user/usage.html#creating-new-release-notes) covering this change. :)

**Review by garrison — APPROVED** [review]
Awesome, thanks!

File: releasenotes/notes/remove-opflow-177bb07ba0e68002.yaml
Comment (caleb-johnson) [inline releasenotes/notes/remove-opflow-177bb07ba0e68002.yaml:1]: Nice

**Pull request merged**
